common/models/change.js

```diff
@@ -186,27 +186,51 @@ module.exports = function(Change) {
 
     cb = cb || utils.createPromiseCallback();
 
-    change.currentRevision(function(err, rev) {
+    var model = this.getModelCtor();
+    var id = this.getModelId();
+
+    model.findById(id, function(err, inst) {
       if (err) return cb(err);
 
-      // avoid setting rev and prev to the same value
-      if (currentRev === rev) {
-        change.debug('rev and prev are equal (not updating anything)');
-        return cb(null, change);
-      }
+      change.tenant = getTenant(model, inst);
```

Comment:
@bajtos 
An issue that has risen with this solution, is that we have to pass the Model instance instead of change instance to the getTenant function. We could do a query for the model instance as we have the model id from the ctx, but this adds another query to each save.

Adding tenant to the change object and calculating the rev, both require the model instance. Our solution gets the model once in rectify function and uses that to set the tenant and revision.